cpp/src/strings/search/find.cu

```diff
@@ -414,6 +414,134 @@ std::unique_ptr<column> contains_warp_parallel(strings_column_view const& input,
+       ++str_byte_idx) {  // iterate the start index in the string
+    for (auto target_idx = 0; target_idx < num_targets; ++target_idx) {  // iterate targets
+      if (!d_results[target_idx][str_idx]) {                             // not found before
+        auto const d_target = d_targets.element<string_view>(target_idx);
+        if (d_str.size_bytes() - str_byte_idx >= d_target.size_bytes() &&
+            (d_target.compare(d_str.data() + str_byte_idx, d_target.size_bytes()) == 0)) {
+          // found
+          d_results[target_idx][str_idx] = true;
+        }
+      }
+    }
+  }
+}
+
+CUDF_KERNEL void multi_contains_warp_parallel_multi_scalars_fn(column_device_view const d_strings,
+                                                               column_device_view const d_targets,
+                                                               cudf::device_span<bool*> d_results)
+{
+  auto const num_targets = d_targets.size();
+  auto const num_rows    = d_strings.size();
+
+  auto const idx    = static_cast<size_type>(threadIdx.x + blockIdx.x * blockDim.x);
+  using warp_reduce = cub::WarpReduce<bool>;
+  __shared__ typename warp_reduce::TempStorage temp_storage;
+
+  if (idx >= (num_rows * cudf::detail::warp_size)) { return; }
+
+  auto const lane_idx = idx % cudf::detail::warp_size;
+  auto const str_idx  = idx / cudf::detail::warp_size;
+  if (d_strings.is_null(str_idx)) { return; }  // bitmask will set result to null.
+
+  // get the string for this warp
+  auto const d_str = d_strings.element<string_view>(str_idx);
+
+  for (size_t target_idx = 0; target_idx < num_targets; target_idx++) {
+    // Identify the target.
+    auto const d_target = d_targets.element<string_view>(target_idx);
+
+    // each thread of the warp will check just part of the string
+    auto found = false;
+    if (d_target.empty()) {
+      found = true;
+    } else {
+      for (auto i = static_cast<size_type>(lane_idx);
+           !found && ((i + d_target.size_bytes()) <= d_str.size_bytes());
+           i += cudf::detail::warp_size) {
+        // check the target matches this part of the d_str data
+        if (d_target.compare(d_str.data() + i, d_target.size_bytes()) == 0) { found = true; }
+      }
+    }
+    __syncwarp();
+    auto const result = warp_reduce(temp_storage).Reduce(found, cub::Max());
+    if (lane_idx == 0) { d_results[target_idx][str_idx] = result; }
+  }
+}
+
+std::vector<std::unique_ptr<column>> multi_contains(strings_column_view const& input,
+                                                    strings_column_view const& targets,
+                                                    bool warp_parallel,
+                                                    rmm::cuda_stream_view stream,
+                                                    rmm::mr::device_memory_resource* mr)
+{
+  auto const num_targets = static_cast<size_type>(targets.size());
+  CUDF_EXPECTS(not targets.is_empty(), "Must specify at least one target string.");
```

Comment:
Recommend adding a check for target nulls too since the logic here is not checking for them.
```
CUDF_EXPECTS(not targets.has_nulls(), "Target strings cannot be null");
```

Reply:
Where did this get added?
Probably should add it to `multi_contains_using_warp_parallel` as well.